src/foam/dao/FUIDAO.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * @license
+ * Copyright 2021 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+foam.CLASS({
+  package: 'foam.dao',
+  name: 'FUIDAO',
+  extends: 'foam.dao.ProxyDAO',
+
+  documentation: `
+    DAO Decorator which sets a specified property\'s value with an global unique ID (generated by UIDGenerator) on DAO.put() 
+    if the value is set to the default value. 
+  `,
+
+  javaImports: [
+    'foam.util.UIDGenerator'
+  ],
+
+  properties: [
+    {
+      class: 'String',
+      name: 'property',
+      value: 'id'
+    },
+    {
+      /** @private */
+      name: 'property_',
```

Comment:
The instance variable for the 'property' property will be called property_, so a different name would cause less conflicts. Like propertyInfo.